src/node_api.cc

```diff
@@ -655,6 +655,36 @@ v8::Local<v8::Object> CreateAccessorCallbackData(napi_env env,
   return cbdata;
 }
 
+// Pointer used to identify items wrapped by N-API. Used by FindWrapper and
+// napi_wrap().
+char napi_wrap_name[] = "N-API Wrapper";
+
+// Search the object's prototype chain for the wrapper object. Usually the
+// wrapper would be the first in the chain, but it is OK for other objects to
+// be inserted in the prototype chain.
+bool FindWrapper(v8::Local<v8::Object> obj,
```

Comment:
If we used a `v8::FunctionTemplate` instead of a simple string we could take advantage of [`v8::Object::FindInstanceInPrototypeChain(Local< FunctionTemplate> tmpl)`](https://v8docs.nodesource.com/node-7.10/db/d85/classv8_1_1_object.html#a0668fdaa32f1ff6d9496e6233d162e7d) instead of having to perform the descent ourselves.

The problem with that is that the `FunctionTemplate` it requires needs to be stored *somewhere*. Now, it can be stored on the `napi_env` but the problem with that is that the env is per-module. So, if one module does `napi_wrap()`, then the other module will not recognize that this has happened, because it will its own copy of a `FunctionTemplate` serving the same purpose. Is this desirable or is this to be avoided?